ads/google/a4a/traffic-experiments.js

```diff
@@ -121,7 +121,16 @@ export function googleAdsIsA4AEnabled(win, element, experimentName,
  */
 function maybeSetExperimentFromUrl(win, experimentName,
     controlBranchId, treatmentBranchId, manualId) {
-  const expParam = parseQueryString(win.location.search)['exp'];
+  let queryString = win.location.search;
+  // If we're inside the AMP viewer, the hash fragment will have been moved
+  // to the originalHash field.
+  const hash = win.location.originalHash || win.location.hash;
```

Comment:
Woo.  Is it really necessary to go to a `Promise` just to get at the hash?  This will be running in the code that decides whether to execute A4A or the existing 3p iframe amp-ad implementation.  I'd prefer not to have to block for an unknown time before making this decision.  Is there a (small, fast) synchronous way to do this?

Also, it looks like `viewer.getFragment()` only accesses `location.hash`, but when I look at AMP docs in the viewer through Chrome dev console, I only see stuff populated in `location.originalHash`.  Looks like `viewer.constructor()` moves it from `hash` to `originalHash`?  So will `getFragment()` even get at that?

But, really, what I want here is "the URL parameter `exp`, whether it's set in the host page URL search fragment or the hash/originalHash fragment".  It looks like `viewer.getParam()` might give me that?  Would that be a simpler way to go here?